src/remote_adapter/remote_adapter_client/remote_client/remote_client.cc

```diff
@@ -0,0 +1,112 @@
+#include "remote_client.h"
+
+#include <iostream>
+
+#include "common/constants.h"
+#include "rpc/detail/log.h"
+
+namespace lua_lib {
+
+RPCLIB_CREATE_LOG_CHANNEL(RemoteClient)
+
+RemoteClient::RemoteClient(connection_ptr connection)
+    : connection_(std::move(connection)) {
+  LOG_INFO("{}", __func__);
+  const int timeout_ = 10000;
+  connection_->set_timeout(timeout_);
+
+  LOG_INFO("Check connection: ");
+
+  auto response =
+      connection_->call(constants::client_connected).as<response_type>();
+
+  if (constants::error_codes::SUCCESS != response.second) {
+    LOG_ERROR("{}", response.first);
+  } else {
+    LOG_INFO("connection OK");
+  }
+}
+
+bool RemoteClient::connected() const {
+  LOG_INFO("{} Check connection:", __func__);
+  if (rpc::client::connection_state::connected ==
+      connection_->get_connection_state()) {
+    LOG_INFO("connection OK");
+    return true;
+  }
+  LOG_ERROR("Not connected");
+  return false;
+}
+
+response_type RemoteClient::file_call(const std::string &rpc_name,
+                                      std::vector<parameter_type> parameters) {
+  LOG_INFO("{0}: of: {1} with {2} parameters", __func__, rpc_name,
+           parameters.size());
+
+  if (connected()) {
+    parameters.push_back(
+        std::make_pair(std::to_string(0), constants::param_types::INT));
+    parameters.push_back(std::make_pair(std::to_string(constants::kMaxSizeData),
+                                        constants::param_types::INT));
+    response_type response =
+        connection_->call(rpc_name, parameters).as<response_type>();
+
+    if (constants::error_codes::FAILED == response.second) {
+      LOG_ERROR("{0}:\nExit Get file data from HU Failed!!!", __func__);
+      return response;
+    }
+
+    std::string tmp_path(constants::kTmpPath);
+    tmp_path.append(parameters[1].first);
+
+    remove(tmp_path.c_str());
+
+    FILE *hFile = fopen(tmp_path.c_str(), "a");
+    if (!hFile) {
+      LOG_ERROR("{0}:\nExit with Failed: \nCan't create file: {1}", __func__,
+                tmp_path);
+      return std::make_pair(std::string("Can't create file"),
+                            constants::error_codes::FAILED);
+    }
+
+    if (constants::error_codes::SUCCESS != response.second) {
```

Comment:
Is this just a way of downloading a file over several calls? In that case, there should be a specific error code to check for, rather than just checking for non-success.

Reply:
@jacobkeeler 
Fixed in:https://github.com/smartdevicelink/sdl_atf/pull/180/commits/14d9eff8e60dab88f5ff3909bf4be368e1a24dd4 https://github.com/smartdevicelink/sdl_atf/pull/180/commits/7aa6965c6d93c67185e572e580e70b3d3698b932 https://github.com/smartdevicelink/sdl_atf/pull/180/commits/f951820274e2b0b0b6f9cc9911da9dae5b69e5f8 https://github.com/smartdevicelink/sdl_atf/pull/180/commits/9c26ee4b2bb87ba04eaea70a6d12fc3551083d48

Reply:
I was thinking that there should be an error code specifically to indicate that there is more data to read.

As it stands, if you were to get an error code such as `NO_CONNECTION`, this would be an infinite loop.

Reply:
@jacobkeeler 
Fixed in:https://github.com/smartdevicelink/sdl_atf/pull/180/commits/7054d5e6a2acba4b723e1f1469cdc9a88b0ce5d9 https://github.com/smartdevicelink/sdl_atf/pull/180/commits/94ae76a5ad6e9e56c3a40bfd23f9207be48d3dca https://github.com/smartdevicelink/sdl_atf/pull/180/commits/bf069cddf81535d5572e255f3cff6015bf5426a1 https://github.com/smartdevicelink/sdl_atf/pull/180/commits/8efce7d247131911cdbba07ea1ecf51efd4cabda

Reply:
I think with using a value separate from a result code in the `response.second` is rather unintuitive. I would suggest making a file in the remote adapter folder which properly documents the structure of these messages for clarity.

In addition, I would suggest the following changes after the latest commits:
```
  offset = (read + offset) == file_len ? error_codes::SUCCESS : read + offset;
```

```
    // A response which includes a non-zero offset indicates that there is more data to read
    if (0 < response.second) {
```

```
  while (remains_bytes > 0);
```

and replace
```
    if (error_codes::FAILED == response.second ||
        error_codes::TIMEOUT_EXPIRED == response.second) {
```
with
```
    if (0 > response.second) {
```
to capture all error codes.

Reply:
@jacobkeeler 
Fixed in:https://github.com/smartdevicelink/sdl_atf/pull/180/commits/38f5feb3ca130db07c441218310fb39cc28261cd and https://github.com/smartdevicelink/sdl_atf/pull/180/commits/1f6e6cae40c375df6733495b16250c411841b42c
@aderiabin 
Please, making a file in the remote adapter folder which properly documents the structure of these messages.

Reply:
@jacobkeeler README file has updated with description of structure of messages related to Remote Testing Adapter server and Remote client library for Lua (see d72c0bbbe34e68af9b18ffb0e9321a15ac5184b4)

Reply:
@aderiabin I was looking for something a bit more detailed, which gives details on the structure of the requests and responses. This is a unique message since it sometimes returns a result which isn't a result code (in this case, an offset is returned), and that should be documented accordingly.

For example, `app_start` has the following

Request: 

1. String app_path - the root path of the app to be started
2. String app_name - the file name of the app to be started

Response: 

1. String result - unused
2. Integer result_code - a predefined result code after performing the operation

Reply:
@jacobkeeler Please find update in d2d5aa0